src/ColorVectorSpace.jl

```diff
@@ -237,6 +238,91 @@ an array: `complement.(x)`
 complement(x::Union{Number,Colorant}) = oneunit(x)-x
 complement(x::TransparentColor) = typeof(x)(complement(color(x)), alpha(x))
 
+"""
+    Complement{T,N,C}
+
+Represents the complementary colorant of its `parent`. To actualize the color,
+`convert` to `C` at which point `complement` will be applied. Like `complement`
+the alpha channel is not modified.
+
+The main interpretation of this type is to invert the *interpretation* of the
+underlying value.
+
+* `Gray(0)` represents black. `Complement(Gray(0))` represents white.
+* `RGB(1,0,0)` represents red. `Complement(RGB(1,0,0))` represents cyan.
+
+Mathematical operations on `Complement` will affect the underlying
+`Colorant` rather than changing the value of the complement. That is
```

Comment:
I'm worried about this behavior---generally, we value the interpretation above the representation. What should `Complement(0.1) + 0.1` return? I'd say 1.0 and not 0.2. `Complement(0.0)` is white, but with this logic, `Complement(0.0) == 1.0` should return `false`.

Reply:
Perhaps `Complement(0.1) + 0.1` should be `Complement(0.0)` then. That is `Complement(x) + y` is `Complement(x-y)`.